sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/DataFeedLiveTests.cs

```diff
@@ -2352,5 +2442,266 @@ private void ValidateSqlServerDataSource(SqlServerDataFeedSource dataSource)
             Assert.That(dataSource.ConnectionString, Is.EqualTo(expectedConnectStr));
             Assert.That(dataSource.Query, Is.EqualTo(DataSourceQuery));
         }
+
+        private void ValidateGenericDataSource(DataFeedSource dataSource, DataFeedSourceType sourceType, bool isAdmin)
```

Comment:
This makes me want to write a reflection based assert to loop through each property of the class and assert that it is not null or empty :)

Reply:
That's an interesting suggestion. Personally I'm not a fan of reflection because it usually makes the code less explicit, and things won't break at compile time.